### PR TITLE
register `TextLayoutInfo` and `TextFlags` type.

### PR DESCRIPTION
**Pull request by ycysdf**
derive `Reflect` to `GlyphAtlasInfo`,`PositionedGlyph` and `TextLayoutInfo`.

# Objective

- I need reflection gets all components of the `TextBundle` and `clone_value` it

## Solution

- registry it

### CODE REVIEW

File: crates/bevy_ui/src/lib.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ pub mod widget;
 
 use bevy_derive::{Deref, DerefMut};
 use bevy_reflect::Reflect;
+use bevy_text::TextLayoutInfo;
```
Comment (nicopap) [inline crates/bevy_ui/src/lib.rs:16]:
```suggestion
#[cfg(feature = "bevy_text")]
use bevy_text::TextLayoutInfo;
```

File: crates/bevy_ui/src/lib.rs [hunk 2]
```diff
@@ -89,6 +91,8 @@ impl Plugin for UiPlugin {
             .register_type::<AlignSelf>()
             .register_type::<BackgroundColor>()
             .register_type::<CalculatedClip>()
+            .register_type::<TextLayoutInfo>()
+            .register_type::<TextFlags>()
```
Comment (nicopap) [inline crates/bevy_ui/src/lib.rs:95]: You need to add those conditionally.

To check your changes, use the `cargo check -p bevy_ui --no-default-features` command

**Comment by nicopap** [comment]
I also need this, I'd like if this got merged. Could you please address the comments?

**Review by nicopap — APPROVED** [review]

**Review by ickshonpe — APPROVED** [review]
This wasn't done before because when I implemented `TextFlags` I was still afraid of touching anything to do with reflection 😅 

Seems good, runs correctly with text disabled. 
Ran the UI example and Text with scale factor is broken again 😓  but it's broken in Main as well so isn't anything to do with this PR.

**Review by mockersf — APPROVED** [review]

**Pull request merged**